ucp/_libs/core.pyx

```diff
@@ -398,10 +406,16 @@ cdef class ApplicationContext:
 
     def __dealloc__(self):
         if self.initiated:
-            for task in self.progress_tasks:
+            # Aborting all objects working in this context
+            for child in (self.children if self.children else []):
```

Comment:
Since it is a `list` type and is initialized to be empty, seems like this should work 🙂

```suggestion
            for child in self.children:
```